pkg/server/endpoints/node/handler.go

```diff
@@ -366,37 +366,43 @@ func (h *Handler) attestToken(ctx context.Context,
 		return nil, errors.New("join token has already been used")
 	}
 
+	tokenValue := string(attestationData.Data)
+
 	ds := h.c.Catalog.DataStores()[0]
-	req := &datastore.JoinToken{Token: string(attestationData.Data)}
-	t, err := ds.FetchToken(ctx, req)
+	resp, err := ds.FetchJoinToken(ctx, &datastore.FetchJoinTokenRequest{
+		Token: tokenValue,
+	})
 	if err != nil {
 		return nil, err
 	}
+	if resp.JoinToken == nil {
+		return nil, errors.New("response missing token")
+	}
+	t := resp.JoinToken
 
 	if t.Token == "" {
 		return nil, errors.New("invalid join token")
 	}
 
+	// Don't fail if we can't delete
+	_, _ = ds.DeleteJoinToken(ctx, &datastore.DeleteJoinTokenRequest{
+		Token: tokenValue,
+	})
```

Comment:
Seeing this diff makes me wonder if we should, in fact, fail if we can't delete following the expiry check. My reasoning here is:
1. If the token we get is expired and can't be deleted, it's ok a future try will still detect that it is stale and try to delete again
1. If the token we get is not expired but also can't be deleted, we will successfully attest the node, and the token will remain valid until its expiration. In many cases, used tokens are leaked into the process table...
1. If the token we get is not expired but also can't be deleted, we can fail the attestation attempt, signaling the user that something went wrong and they should retry

I guess it's out of scope for this change - just something that popped up

Reply:
I agree deleting is the right choice. I'm happy to make that change.